doc/guidelines/frontend-keyhandling.md

```diff
@@ -0,0 +1,80 @@
+# Frontend Keybindings
+
+This document gives guidelines on how to implement keybindings in the frontend.
+Our key handling solution is designed to address the following key constraints:
+
+- **Developer-friendly**: Developers should get an overview where which key binding is assigned.
+- **UX**: Collisions between shortcuts which trigger multiple actions at once should be avoided.
+
+### Important notes for Konva Key handling
+
+- since keys cannot be bound directly on konva elements, keyhandlers have to check if the corresponding layer is active to avoid collisions
+- if keys should only be active if map is focused, handlers have to be bound to canvas section of the map component
+
+### Keybinding Configuration
+
+The file 'keybindings.json' contains all keybindings that are used in the application and serves the following characteristics:
+
+- keybindings can easily be changed by editing the file
+- developers have an overview of all keybindings.
+- there are different sections for keybindings:
+  - global: keybindings that are globally active
+  - scopes for specific layers: keybindings that are only active if the corresponding layer is active. this scopes should be named according to the layer name
+  - _just_for_documentation_: keybindings that are only used for documentation purposes and are not meant to get configured
+- modifier keys are supported. (e.g. Ctrl+k)
+- multiple keybindings can be assigned to one action
+
+Structure:
+
+```
+{
+  "global": {
+    "<<action_name>>": ["<<Key-Binding>>, ...]"
```

Comment:
I'd suggest swapping the key and the value here if it is not too much effort:
"<<key_binding>>": "<<action_name>>"
The reason is that I think it is easier to look up the corresponding action within the event handler: e.g. config.global['Escape'] to get the corresponding action.
It is also easier to check if there are keybindings assigned multiple times as the json is not valid anymore.

Reply:
I also suggested that already but everyone else found it more clear as it is proposed now. A check if key-bindings are used several times should be quite simple in both cases.

> config.global['Escape']

This looks to me to a reason to keep it as proposed. The code shouldn't bother about which keybinding is used, otherwise you couldn't simply make changes in the JSON. Or did I misunderstand?

Reply:
Fair enough. I don't have a strong opinion about it, it is just more intuitive for me as in my experience keybindings are often defined with binding -> action in my experience.
For example vim config, sxhkd config but that doesn't necessarily mean that it is better for us :)
It is not hard to check for duplicates in the config file but my point was that you have to do it.
When the keybinding is the key we have an invalid json as soon as the keybinding is used twice.
On the other hand it might be quicker to look up the bindings from the config with the action as key so we can leave it like that :)